src/fabric.rs

```diff
@@ -14,40 +14,41 @@ use crate::interval::{Interval, Role, Material};
 use crate::interval::Role::{Measure, Pull, Push};
 use crate::interval::Span::{Approaching, Fixed};
 use crate::joint::Joint;
+use crate::physics::Physics;
 use crate::tenscript::Spin;
-use crate::world::World;
 
-#[derive(Clone, Debug, PartialEq)]
+#[derive(Clone, Default, Debug, PartialEq)]
 pub struct Progress {
     limit: usize,
     count: usize,
 }
 
-impl Default for Progress {
-    fn default() -> Self {
-        Self { count: 1, limit: 1 }
-    }
-}
-
 impl Progress {
     pub fn start(&mut self, countdown: usize) {
         self.count = 0;
         self.limit = countdown;
     }
 
-    pub fn step(&mut self) {
+    fn step(&mut self) -> bool { // true if it takes the final step
         let count = self.count + 1;
         if count <= self.limit {
             self.count = count;
+            self.count == self.limit // final step?
+        } else {
+            false
         }
```

Comment:
Matter of preference

```suggestion
        if count > self.limit {
            return false;
        }
        self.count = count;
        self.count == self.limit // final step?
```